src/Microsoft.AspNetCore.Mvc.Core/Routing/DispatcherUrlHelper.cs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Mvc.Routing
+{
+    /// <summary>
+    /// An implementation of <see cref="IUrlHelper"/> that uses <see cref="ILinkGenerator"/> to build URLs 
+    /// for ASP.NET MVC within an application.
+    /// </summary>
+    public class DispatcherUrlHelper : IUrlHelper
```

Comment:
Does this need to be public?

Reply:
Sure, makes sense to not make it public.